File: lib/types/price_update.ex
defmodule Types.PriceUpdate do
  @moduledoc """
  Price update struct
  """

  @enforce_keys [
    :symbol,
    :ask_price,
    :ask_qty,
    :bid_price,
    :bid_qty
  ]
  defstruct [
    :symbol,
    :ask_price,
    :ask_qty,
    :bid_price,
    :bid_qty
  ]

  @type t :: %__MODULE__{
          symbol: String.t(),
          ask_price: Decimal.t(),
          ask_qty: Decimal.t(),
          bid_price: Decimal.t(),
          bid_qty: Decimal.t()
        }
end

File: src/PriceUpdate.cpp
#include "PriceUpdate.h"

/**
 * @brief Stream operator overload for PriceUpdate
 * @param os Output stream
 * @param update PriceUpdate to print
 * @return Reference to the output stream
 */
auto operator<<(std::ostream &os, const PriceUpdate &update) -> std::ostream & {
  os << "PriceUpdate{"
     << "symbol='" << update.symbol << "', "
     << "bid=" << update.bidPrice << "@" << update.bidQty << ", "
     << "ask=" << update.askPrice << "@" << update.askQty << "}";
  return os;
}